src/app/updater.js

```diff
@@ -77,21 +79,44 @@
                 self.updateData = updateData;
                 return true;
             }
+            if (App.settings.UpdateSeed) {
+                client.add(updateData.UpdateUrl, { path: os.tmpdir() }, function (torrent) {
+                    torrent.on('error', function (err) {
+                        win.debug('ERROR' + err.message);
+                    });
+                    torrent.on('done', function () {
+                        win.debug('Seeding the Current Update!');
+                    });
+                });
+            }
 
             win.debug('Not updating because we are running the latest version');
             return false;
         });
     };
-
-    Updater.prototype.download = function (source, output) {
+    
+    
+    Updater.prototype.download = function (source, outputDir) {
         var defer = Q.defer();
-        var downloadStream = request(source);
-        win.debug('Downloading update... Please allow a few minutes');
-        downloadStream.pipe(fs.createWriteStream(output));
-        downloadStream.on('complete', function () {
-            win.debug('Update downloaded!');
-            defer.resolve(output);
+        client.on('error', function (err) {
+            win.debug('ERROR: ' + err.message);
+            defer.reject(err);
         });
+
+        client.add(source, {
+            path: outputDir
+        }, function (torrent) {
+            win.debug('Downloading update... Please allow a few minutes');
+            torrent.on('error', function (err) {
+                win.debug('ERROR' + err.message);
+                defer.reject(err);
+            });
+            torrent.on('done', function () {
+                win.debug('Update downloaded!');
+                defer.resolve(path.join(outputDir, torrent.name));
```

Comment:
torrent.name contains the extension of the file in this case? Because there's a risk it's a directory